week1/intro.cpp

```diff
@@ -0,0 +1,27 @@
+class Foo {
+public:
+    std::mutex two,three;
```

Comment:
Nit:
Your solution seems to work well for this problem, but having a thread-specific mutex might not be a good idea for most such problems, because it doesn't scale.
Again, feel free to ignore my comment, because you're doing what is asked here, but I think we should start practicing "right" patterns.